packages/common/src/doc-cache.ts

```diff
@@ -0,0 +1,69 @@
+import { LRUMap } from 'lru_map'
+import DocID from "@ceramicnetwork/docid"
+
+import { DocStateHolder } from "./doctype"
+
+/**
+ * Encapsulates document caching (base, commits)
+ */
+export class DocCache {
+    private readonly _cacheCommits: boolean
+    private readonly _baseCache: LRUMap<string, DocStateHolder>
+    private readonly _commitCache: LRUMap<string, DocStateHolder>
+
+    constructor(limit, cacheCommits = true) {
+        this._cacheCommits = cacheCommits
+        this._baseCache = new LRUMap(limit)
+
+        // use the same 'limit' if cacheCommits is enabled
+        this._commitCache = this._cacheCommits? new LRUMap(limit) : new LRUMap(0)
+    }
+
+    /**
+     * Sets to cache
+     * @param doc - DocStateHolder instance
+     */
+    set(doc: DocStateHolder): void {
+        this._baseCache.set(doc.id.baseID.toString(), doc)
+        if (this._cacheCommits && doc.id.commit) {
+            this._commitCache.set(doc.id.toString(), doc)
+        }
+    }
```

Comment:
Not sure about the logic in this method. Seems like if I cache an older commit it will also be cached as the `_baseCache`?

Reply:
I guess the baseCache should only be set if `doc.id.commit` is not set?

Reply:
yep, my bad 👍